core/src/test/java/org/elasticsearch/index/query/MultiMatchQueryBuilderTests.java

```diff
@@ -0,0 +1,185 @@
+
+        multiMatchQueryBuilder = new MultiMatchQueryBuilder(null);
+        assertThat(multiMatchQueryBuilder.validate().validationErrors().size(), is(2));
+
+        multiMatchQueryBuilder = new MultiMatchQueryBuilder("text", "field");
+        assertNull(multiMatchQueryBuilder.validate());
+    }
+
+    @Override
+    protected void assertBoost(MultiMatchQueryBuilder queryBuilder, Query query) throws IOException {
+        //we delegate boost checks to specific boost tests below
+    }
+
+    @Test
+    public void testToQueryBoost() throws IOException {
+        assumeTrue("test runs only when at least a type is registered", getCurrentTypes().length > 0);
+        QueryShardContext shardContext = createShardContext();
+        MultiMatchQueryBuilder multiMatchQueryBuilder = new MultiMatchQueryBuilder("test");
+        multiMatchQueryBuilder.field(STRING_FIELD_NAME, 5);
+        Query query = multiMatchQueryBuilder.toQuery(shardContext);
+        assertThat(query, instanceOf(TermQuery.class));
+        assertThat(query.getBoost(), equalTo(5f));
+
+        multiMatchQueryBuilder = new MultiMatchQueryBuilder("test");
+        multiMatchQueryBuilder.field(STRING_FIELD_NAME, 5);
+        multiMatchQueryBuilder.boost(2);
+        query = multiMatchQueryBuilder.toQuery(shardContext);
+        assertThat(query, instanceOf(TermQuery.class));
+        assertThat(query.getBoost(), equalTo(10f));
+    }
+
+    @Test
+    public void testToQueryMultipleTermsBooleanQuery() throws Exception {
+        assumeTrue("test runs only when at least a type is registered", getCurrentTypes().length > 0);
+        Query query = multiMatchQuery("test1 test2").field(STRING_FIELD_NAME).useDisMax(false).toQuery(createShardContext());
+        assertThat(query, instanceOf(BooleanQuery.class));
+        BooleanQuery bQuery = (BooleanQuery) query;
+        assertThat(bQuery.clauses().size(), equalTo(2));
+        assertThat(assertBooleanSubQuery(query, TermQuery.class, 0).getTerm(), equalTo(new Term(STRING_FIELD_NAME, "test1")));
+        assertThat(assertBooleanSubQuery(query, TermQuery.class, 1).getTerm(), equalTo(new Term(STRING_FIELD_NAME, "test2")));
+    }
+
+    @Test
+    public void testToQueryMultipleFieldsBooleanQuery() throws Exception {
+        assumeTrue("test runs only when at least a type is registered", getCurrentTypes().length > 0);
+        Query query = multiMatchQuery("test").field(STRING_FIELD_NAME).field(STRING_FIELD_NAME_2).useDisMax(false).toQuery(createShardContext());
+        assertThat(query, instanceOf(BooleanQuery.class));
+        BooleanQuery bQuery = (BooleanQuery) query;
+        assertThat(bQuery.clauses().size(), equalTo(2));
+        assertThat(assertBooleanSubQuery(query, TermQuery.class, 0).getTerm(), equalTo(new Term(STRING_FIELD_NAME, "test")));
+        assertThat(assertBooleanSubQuery(query, TermQuery.class, 1).getTerm(), equalTo(new Term(STRING_FIELD_NAME_2, "test")));
+    }
```

Comment:
Nice, I like it to have some more controlled, non randomized tests here for basic behaviour.